symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/message/util/MessageUtil.java

```diff
@@ -0,0 +1,81 @@
+package com.symphony.bdk.core.service.message.util;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.apiguardian.api.API;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Helper class for pre-processing an outgoing message
+ */
+@API(status = API.Status.EXPERIMENTAL)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class MessageUtil {
+  private static Map<String, String> tokens = initializeMap();
+  private static final Pattern PATTERN = initializePattern(tokens);
+
+  /**
+   * This method takes care of all special characters placed within the messageML that must be HTML-escaped
+   * to have a valid MessageML format.
+   *
+   * @param rawText text to be parsed
+   * @return text in a valid messageML format
+   */
+  public static String escapeSpecialChars(String rawText) {
+    Matcher matcher = PATTERN.matcher(rawText);
+    StringBuffer parsedText = new StringBuffer();
+    while (matcher.find()) {
+      matcher.appendReplacement(parsedText, getReplacementOf(matcher.group(1)));
+    }
+    matcher.appendTail(parsedText);
+    return parsedText.toString();
+  }
+
+  private static String getReplacementOf(String group) {
+    String replacement = tokens.get(group);
+    if (replacement == null) {
+      replacement = tokens.get("\\" + group);
+    }
+    return replacement;
+  }
+
+  public static String escape(String text) {
```

Comment:
as a user should I use escapeSpecialChars or escape? (something we could clarify with comments maybe?)

Reply:
sorry I pushed that by mistake, it was part of my tests to use the StringEscapeUtils but it is not working for every character. I will remove this function since it is working only partially.